### DISCUSSION TITLE
Model Trainer was not set before constructing the dataset, incorrect number of training batches will be used. Please set the trainer and rebuild the dataset.

### DISCUSSION THREAD

**Discussion by mehadi92** [NVIDIA-NeMo/NeMo#7471] [Q&A] [answered]
I'm getting this although I set
```
model.setup_training_data(cfg.train_ds)
model.setup_multiple_validation_data(cfg.validation_ds)
model.setup_multiple_test_data(cfg.test_ds)
```
Is there anything that I'm missing

Thanks

**Comment by titu1994** [comment, marked as answer]
You need to call model.set_trainer() before calling the data loader setup methods. Without knowing the training env (number of GPUs, nodes) it is not possible to pre calculate the number of training batches across the environment.
